### PR TITLE
feat: enable CSP for SSG

### PR DESCRIPTION
**Pull request by tresko**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Add Content-Security-Policy meta tag for SSG pages.

Enables Content Security Policy for SSG pages.

Resolves: #108 

## Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes (if not applicable, please state why)

### CODE REVIEW

**Comment by Baroshem** [comment]
Hey @tresko 

Thank you so much for this PR. I really like the idea! I will review it tomorrow and provably recommend you to write some docs and configuration so that this nitro plugin would be enabled automatically if the app is generated as SSG.

Stay tuned!

File: src/runtime/nitro.ts [hunk 1]
```diff
@@ -1,12 +1,132 @@
...
+  nitro.hooks.hook('render:response', (response: RenderResponse, { event }: { event: H3Event }) => {
     // Temporary as in Nuxt 3.0.0 header name is 'X-Powered-By' and in 3.1.X is 'x-powered-by'
     if (response.headers['x-powered-by']) {
       delete response.headers['x-powered-by']
     } else if (response.headers['X-Powered-By']) {
       delete response.headers['X-Powered-By']
     }
   })
+
+  nitro.hooks.hook('render:html', (html: NuxtRenderHTMLContext, { event }: { event: H3Event }) => {
```
Comment (Baroshem) [inline src/runtime/nitro.ts:33]: I would recommend creating a new file where this custom plugin could be registered as currently it is registered as a part of a XPoweredBy plugin so once a user disabled hidePoweredBy, this hook will not be triggered as well.

What is more, I would enable this module without any configuration if the project is running in SSG by default and  remain configurability from `security.headers.contentSecurityPolicy.value`

File: src/runtime/nitro.ts [hunk 2]
```diff
@@ -1,12 +1,132 @@
...
+      } else if (typeof value === 'boolean') {
+        policyValue = ''
+      } else {
+        policyValue = value
+      }
+
+      contentArray.push(`${key} ${policyValue}`)
+    }
+    const content = contentArray.join('; ')
+    console.log(content)
```
Comment (Baroshem) [inline src/runtime/nitro.ts:90]: This can be deleted ;)

**Comment by Baroshem** [comment]
Also, I would add an appriopriate documentation section about this SSG functionality. 

Probably a new heading in the setup -> https://nuxt-security.vercel.app/getting-started/setup just above configuration called for example `SSG Apps`:

`This module is meant to work with SSR apps but you can also use this module in SSG apps where you will get a  CSP support`

**Comment by Baroshem** [comment]
Let me know @tresko if you have time to include these changes. If not, I can do that, no worries. I will mention you anyway in the release and in social media as the author of this feature and the contributor :)

**Comment by tresko** [comment]
No problem, I will add it.

**Comment by tresko** [comment]
Done, can you check it? :)

**Comment by Baroshem** [comment]
Nicely done @tresko 💚

I am now merging this feature and going to test it out in more details :)

Thank you so much for the contribution. Feel free to recommend other features that will make this module deliver even better experience :)

**Pull request merged**